test/e2e/vsphere_volume_vsan_policy.go

```diff
@@ -0,0 +1,281 @@
+	It("verify VSAN storage capability diskStripes - 1 and objectSpaceReservation - 30 and datastore - sharedVmfs-0 is not honored for dynamically provisioned pvc using storageclass", func() {
+		By("Invoking Test for VSAN policy diskStripes: 1, objectSpaceReservation: 30 and a non-VSAN datastore: sharedVmfs-0")
+		scParameters[Policy_DiskStripes] = "1"
+		scParameters[Policy_ObjectSpaceReservation] = "30"
+		scParameters[Datastore] = VmfsDatastore
+		framework.Logf("Invoking Test for VSAN storage capabilities: %+v", scParameters)
+		err := invokeInvalidVSANPolicyTestNeg(client, namespace, scParameters)
+		framework.Logf("Yo: %+q", err.Error())
+		errorMsg := "The specified datastore: \\\"" + VmfsDatastore + "\\\" is not a VSAN datastore. " +
+			"The policy parameters will work only with VSAN Datastore."
+		framework.Logf("errorMsg: %+q", errorMsg)
+		if !strings.Contains(err.Error(), errorMsg) {
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+})
+
+func invokeValidVSANPolicyTest(client clientset.Interface, namespace string, scParameters map[string]string) {
+	By("Creating Storage Class With VSAN policy params")
+	storageClassSpec := getVSphereStorageClassSpec("vsanpolicysc", scParameters)
+	storageclass, err := client.StorageV1beta1().StorageClasses().Create(storageClassSpec)
+	if err != nil {
+		framework.Logf("Failed to create storage class with err: %+v", err)
+	}
+	Expect(err).NotTo(HaveOccurred())
+
+	defer client.StorageV1beta1().StorageClasses().Delete(storageclass.Name, nil)
+
+	By("Creating PVC using the Storage Class")
+	pvclaimSpec := getVSphereClaimSpecWithStorageClassAnnotation(namespace, storageclass)
+	pvclaim, err := client.CoreV1().PersistentVolumeClaims(namespace).Create(pvclaimSpec)
+	if err != nil {
+		framework.Logf("Failed to create PVC with err: %+v", err)
+	}
+	Expect(err).NotTo(HaveOccurred())
+
+	defer func() {
+		client.CoreV1().PersistentVolumeClaims(namespace).Delete(pvclaimSpec.Name, nil)
+	}()
+
+	By("Waiting for claim to be in bound phase")
+	err = framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client, pvclaim.Namespace, pvclaim.Name, framework.Poll, framework.ClaimProvisionTimeout)
+	if err != nil {
+		framework.Logf("Failed to bound PVC with err: %+v", err)
+	}
+	Expect(err).NotTo(HaveOccurred())
+
+	// Get new copy of the claim
+	pvclaim, err = client.CoreV1().PersistentVolumeClaims(pvclaim.Namespace).Get(pvclaim.Name, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+
+	// Get the bound PV
+	pv, err := client.CoreV1().PersistentVolumes().Get(pvclaim.Spec.VolumeName, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+
+	/*
+	   PV is required to be attached to the Node. so that using govmomi API we can grab Disk's Backing Info
+	   to check EagerlyScrub and ThinProvisioned property
+	*/
+	By("Creating pod to attach PV to the node")
+	// Create pod to attach Volume to Node
+	podSpec := getVSpherePodSpecWithClaim(pvclaim.Name, nil, "while true ; do sleep 2 ; done")
```

Comment:
framework.CreateClientPod() might work for you depending on the podspec you need.